Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -190,6 +198,25 @@ protected async virtual Task<ChannelCredentials> GetChannelCredentialsAsync(Canc
         /// </summary>
         protected abstract ServiceEndpoint GetDefaultEndpoint();
 
+        /// <summary>
+        /// Returns the channel pool to use when no other options are specified. This method is not called unless
+        /// <see cref="CanUseChannelPool"/> returns true, so if a channel pool is unavailable, override that property
+        /// to return false and throw an exception from this method.
+        /// </summary>
+        protected abstract ChannelPool GetChannelPool();
+
+        /// <summary>
+        /// Returns whether or not a channel pool can be used if a channel is required. The default behavior is to return
+        /// true if and only if no scopes, credentials or token access method have been specified. Derived classes should
+        /// override this property if there are other reasons why the channel pool should not be used.
+        /// </summary>
+        protected virtual bool CanUseChannelPool =>
+            ChannelCredentials == null &&
+            CredentialsPath == null &&
+            JsonCredentials == null &&
+            TokenAccessMethod == null &&
```

Comment:
TokenAccessMethod is not used anywhere that I can see, just in validations?

Reply:
Yikes, you're right. It should basically be used to create credentials if necessary.